File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_nums = {}

        for i in range(len(nums)):
            key = target - nums[i]
            if dict_nums.get(key) is not None:
                return [i, dict_nums[key]]
            dict_nums[nums[i]] = i

File: 0001-two-sum/0001-two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numIndex = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (numIndex.get(target - nums[i]) == null) {
                numIndex.put(nums[i], i);
            } else {
                return new int[]{numIndex.get(target-nums[i]), i};
            }
        }
        return null;
    }
}